pkg/minishift/openshift/oc.go

```diff
@@ -68,3 +90,114 @@ func AddContextForProfile(profile string, ip string, username string, namespace
 	}
 	return nil
 }
+
+// Get the route for service
+func GetServiceURL(service, namespace string, https bool) (string, error) {
+	urlScheme := "http://"
+	if https {
+		urlScheme = "https://"
+	}
+
+	if namespace == "default" {
+		return "", errors.New("Namespace need to be specified with -n option")
+	}
+
+	if !isProjectExists(namespace) {
+		return "", errors.New(fmt.Sprintf("Namespace %s doesn't exits", namespace))
+	}
+
+	cmdName := instanceState.Config.OcPath
+	cmdArgText := fmt.Sprintf("get route/%s -n %s --config=%s %s", service, namespace, systemKubeConfigPath, URLCustomCol)
+	tokens := strings.Split(cmdArgText, " ")
+	cmdOut, err := runner.Output(cmdName, tokens...)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Route not found for '%s'", service))
+	}
+
+	url := strings.Split(string(cmdOut), "\n")[1] // second element contain actual URL content
+	return urlScheme + url, nil
+}
+
+// Get the available routes to user
+func GetServiceURLs(serviceListNamespace string) ([]ServiceURL, error) {
+	var namespaces []string
+	var err error
+	var serviceURLs []ServiceURL
+
+	if serviceListNamespace != "" && !isProjectExists(serviceListNamespace) {
+		return serviceURLs, errors.New(fmt.Sprintf("Namespace %s doesn't exits", serviceListNamespace))
+	}
+
+	// If namespace is empty then consider all namespaces user belongs to
+	if serviceListNamespace == "" {
+		namespaces, err = getProjects()
+		if err != nil {
+			return serviceURLs, errors.New(fmt.Sprintf("Error getting service urls ", err))
+		}
+	} else {
+		namespaces = append(namespaces, serviceListNamespace)
+	}
+
+	// iterate over projects, get route and store into ServieURL format
+	for _, namespace := range namespaces {
+		cmdName := instanceState.Config.OcPath
+		cmdArgText := fmt.Sprintf("get route -n %s --config=%s %s", namespace, systemKubeConfigPath, URLsCustomCol)
+		tokens := strings.Split(cmdArgText, " ")
+		cmdOut, err := runner.Output(cmdName, tokens...)
+		if err != nil {
+			return serviceURLs, err
+		}
+
+		results := string(cmdOut) // convert into string
```

Comment:
This could be nicely made a small helper function. Splitting tis up allows you test this in isolation. Maybe it can even be a generic string utility method which we can use at other places. 

this is in general a very long method. Too long imo. This could easily be split onto three or four method, making the code easier to read and test. The various loops are probably good candidates to extract.